File: backend/byteBlog/src/main/java/com/jeswin8801/byteBlog/entities/dto/comment/PostCommentDto.java
package com.jeswin8801.byteBlog.entities.dto.comment;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostCommentDto {

    @JsonProperty(required = true)
    private String comment;

    @JsonProperty(value = "user-id", required = true)
    private String userId;

    @JsonProperty(value = "blog-id", required = true)
    private String blogId;

    // can be null as the first comments associated with a blog has no parent comment and the Blog is the parent
    @JsonProperty("parent-comment-id")
    private Long parentCommentId;
}

File: frontend/src/app/models/dtos/comment/post-comment-dto.ts
import { JsonProperty } from 'json-object-mapper';

export class PostCommentDto {
  comment?: string;

  @JsonProperty({ name: 'user-id' })
  userId?: string;

  @JsonProperty({ name: 'blog-id' })
  blogId?: string;

  @JsonProperty({ name: 'parent-comment-id' })
  parentCommentId?: number;
}
